packages/components/src/local/mapping/index.tsx

```diff
@@ -252,7 +233,9 @@ export const Mapping: React.FC<PropTypes> = ({
         for (let ctr = 0; ctr < plannedTurn.turnsVal; ctr++) {
           const step: RouteStep = { turn: ++turnStart, status: { state: status.name } }
           // store this step
-          store = routeAddStep(store, selectedAsset.uniqid, step)
+          if (selectedAsset) {
+            store = routeAddStep(store, selectedAsset.uniqid, step)
```

Comment:
Be careful with circular references like this.  You might be safer doing `updatedStore = routeAddStep(etc...)` obviously don't forget to declare `updateStore` before you reference it.

Reply:
```suggestion
           // before conditional 
           let updatedStore
           
           // in conditional
            updatedStore = routeAddStep(store, selectedAsset.uniqid, step)
            
            // afterConditional
            setRouteStore(updatedStore)
```

Reply:
In the loop, we need to add the new step to the modified store:

 updatedStore = routeAddStep(***updatedStore***, selectedAsset.uniqid, step)

Reply:
Leave it as it is then if that's the way it has to be. Maybe rename the variable though as there is a constant with the same name earlier in the file and it was momentarily confusing.